astropy/coordinates/angles/_wrap.pyx

```diff
@@ -0,0 +1,55 @@
+#cython: language_level=3
+cimport cython
+cimport numpy as np
+from libc cimport math
+
+np.import_array()
+
+ctypedef fused dtype:
+    short
+    int
+    long
+    float
+    double
+
+
+@cython.ufunc
+@cython.cdivision(True)
+cdef dtype wrap_at(dtype angle, dtype wrap_angle, dtype period):
+    cdef dtype wrap_angle_floor
+    wrap_angle_floor = wrap_angle - period
+
+    if dtype is float or dtype is double:
```

Comment:
Just out of curiosity as I haven't seen the cython ufunc stuff before, are these checks going to be done for every single value? Or does this get optimised out?

Reply:
Checks on fuses dtypes are essentially like `if constexpr`, the function is compiled for each member of the fused type separately

Reply:
http://docs.cython.org/en/latest/src/userguide/fusedtypes.html#type-checking-specializations